pallets/liquidity-pools-gateway/src/message_processing.rs

```diff
@@ -333,7 +334,11 @@ impl<T: Config> Pallet<T> {
 				// we can return.
 				None => return Ok(()),
 				Some(stored_inbound_entry) => match stored_inbound_entry {
-					InboundEntry::Message(message_entry) => message = Some(message_entry.message),
+					InboundEntry::Message(message_entry)
+						if message_entry.session_id == session_id =>
```

Comment:
Question: If `session_id` is different, should we remove the entry?

Reply:
We should not because then it's impossible to replay the message and funds are stuck on the EVM side. Keeping entries for an old session id can be made unstuck via `execute_message_recovery`.

Reply:
> Question: If `session_id` is different, should we remove the entry?

We should not because then it's impossible to replay the message and funds are stuck on the EVM side. Keeping entries for an old session id can be made unstuck via `execute_message_recovery`.

Reply:
`execute_message_recovery` will only increase the proof count for a specific router ID, we will still hit this logic and be unable to execute a message from an older session. Maybe we should extend `execute_message_recovery` and either:
- set the session of a message entry to the current one;
- increase the proof count - current behavior;

Reply:
Bumping this again since I think we might need to add the 1st point that I mentioned above.

cc @hieronx